pointcloud/basic_octree.go

```diff
@@ -182,3 +189,158 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
+			octree.label,
+			octree.meta,
+		}
+		transformedOctree.center = spatialmath.Compose(octree.center, p)
+		transformPoint := p.Point()
+
+		transformedOctree.meta.MaxX += transformPoint.X
+		transformedOctree.meta.MinX += transformPoint.X
+		transformedOctree.meta.MaxY += transformPoint.Y
+		transformedOctree.meta.MinY += transformPoint.Y
+		transformedOctree.meta.MaxZ += transformPoint.Z
+		transformedOctree.meta.MinZ += transformPoint.Z
+
+		transformedOctree.meta.totalX = 0
+		transformedOctree.meta.totalY = 0
+		transformedOctree.meta.totalZ = 0
+
+		for _, child := range octree.node.children {
+			transformedChild := child.Transform(p).(*BasicOctree)
+			transformedOctree.node.children = append(transformedOctree.node.children, transformedChild)
+
+			transformedOctree.meta.totalX += transformedChild.meta.totalX
+			transformedOctree.meta.totalY += transformedChild.meta.totalY
+			transformedOctree.meta.totalZ += transformedChild.meta.totalZ
+		}
+	case leafNodeEmpty:
+		transformedOctree = &BasicOctree{
+			basicOctreeNode{
+				octree.node.nodeType,
+				octree.node.children,
+				octree.node.point,
+				octree.node.maxVal,
+			},
+			spatialmath.Compose(octree.center, p),
+			octree.sideLength,
+			octree.size,
+			octree.label,
+			octree.meta,
+		}
+	case leafNodeFilled:
+		transformedOctree = &BasicOctree{
+			basicOctreeNode{
+				octree.node.nodeType,
+				octree.node.children,
+				octree.node.point,
+				octree.node.maxVal,
+			},
+			octree.center,
+			octree.sideLength,
+			octree.size,
+			octree.label,
+			octree.meta,
+		}
+		transformPoint := p.Point()
+		transformedOctree.center = spatialmath.Compose(octree.center, p)
+		transformedOctree.node.point = &PointAndData{P: octree.node.point.P.Add(transformPoint), D: octree.node.point.D}
+
+		transformedOctree.meta.MaxX += transformPoint.X
+		transformedOctree.meta.MinX += transformPoint.X
+		transformedOctree.meta.MaxY += transformPoint.Y
+		transformedOctree.meta.MinY += transformPoint.Y
+		transformedOctree.meta.MaxZ += transformPoint.Z
+		transformedOctree.meta.MinZ += transformPoint.Z
+
+		transformedOctree.meta.totalX += transformPoint.X
+		transformedOctree.meta.totalY += transformPoint.Y
+		transformedOctree.meta.totalZ += transformPoint.Z
+	}
+	return transformedOctree
+}
+
+// ToProtobuf converts the octree to a Geometry proto message.
+func (octree *BasicOctree) ToProtobuf() *commonpb.Geometry {
+	// TODO
+	return nil
+}
+
+// CollidesWith checks if the given octree collides with the given geometry and returns true if it does.
+func (octree *BasicOctree) CollidesWith(geom spatialmath.Geometry) (bool, error) {
+	return octree.CollidesWithGeometry(geom, confidenceThreshold, buffer)
+}
+
+// DistanceFrom returns the distance from the given octree to the given geometry.
+func (octree *BasicOctree) DistanceFrom(geom spatialmath.Geometry) (float64, error) {
+	// TODO: currently implemented as the bare minimum but needs to be changed to correct implementation
```

Comment:
TODO in this PR or as a future JIRA Ticket?

Reply:
future ticket probably

Reply:
Yeah if you could ticket this before closing that would be great! (I can help with that)